### PR TITLE
Added docstrings D102 to public method on Bio.SeqIO

### PR DESCRIPTION
**Pull request by svalqui**
This pull request addresses issue #1963 

- [X] I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

- [X] I have read the ``CONTRIBUTING.rst`` file and understand that AppVeyor and
TravisCI will be used to confirm the Biopython unit tests and ``flake8`` style
checks pass with these changes.

- [X] I have added my name to the alphabetical contributors listings in the files
``NEWS.rst`` and ``CONTRIB.rst`` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

### CODE REVIEW

File: Bio/SeqIO/_index.py [hunk 1]
```diff
@@ -39,6 +39,8 @@
 
 
 class SeqFileRandomAccess(_IndexedSeqFileProxy):
+    """Random Access File generic."""
```
Comment (peterjc) [inline Bio/SeqIO/_index.py]: How about something more like this?:

``Base class for defining random access to sequence files.``

File: Bio/SeqIO/_index.py [hunk 2]
```diff
@@ -140,6 +142,7 @@ def __iter__(self):
         SeqIO.SffIO._check_eof(handle, index_offset, index_length)
 
     def get(self, offset):
+        """Return the Sequence records at the given offset."""
```
Comment (peterjc) [inline Bio/SeqIO/_index.py]: Please avoid the term "Sequence record" in favour of "SeqRecord", also this should be singular here not plural, how about:

```
    """Return the SeqRecord starting at the given offset."""
```

File: Bio/SeqIO/_index.py [hunk 3]
```diff
@@ -156,7 +159,10 @@ def get_raw(self, offset):
 
 
 class SffTrimedRandomAccess(SffRandomAccess):
+    """Random access to a SFF file with sequence records trimmed."""
```
Comment (peterjc) [inline Bio/SeqIO/_index.py]: How about:

```
"""Random access to an SFF file with defined trimming applied to each sequence."""
```

File: Bio/SeqIO/_index.py (hunk 2 above)
Comment (peterjc) [reply Bio/SeqIO/_index.py]: (Same applies to the other ``get`` methods in this file)

File: Bio/SeqIO/Interfaces.py [hunk 4]
```diff
@@ -175,12 +175,20 @@ def __init__(self, handle):
         self._footer_written = False
 
     def write_header(self):
+        """Set _header_written to True.
```
Comment (peterjc) [inline Bio/SeqIO/Interfaces.py]: This is not helpful. How about:

```
    """Default method for writing file footer.

    If your file format defines a footer, you should implement this method
    in order to write the footer after all the records.

    The default implementation checks the private attribute ._header_written
    to ensure the footer is only written once.
    """"

File: Bio/SeqIO/Interfaces.py [hunk 5]
```diff
@@ -175,12 +175,20 @@ def __init__(self, handle):
...
+
+        To control the header is written only once.
+        """
         assert not self._header_written, "You have aleady called write_header()"
         assert not self._record_written, "You have aleady called write_record() or write_records()"
         assert not self._footer_written, "You have aleady called write_footer()"
         self._header_written = True
 
     def write_footer(self):
+        """Set _footer_written to True.
```
Comment (peterjc) [inline Bio/SeqIO/Interfaces.py]: This is not helpful. How about:

```
    """Default method for writing file footer.

    If your file format defines a footer, you should implement this method
    in order to write the footer after all the records.

    The default implementation checks the private attribute ._footer_written
    to ensure the footer is only written once.
    """"
```

File: Bio/SeqIO/Interfaces.py (hunk 4 above)
Comment (svalqui) [reply Bio/SeqIO/Interfaces.py]: 'Default' at the begin of the DocString returns "D401 First line should be in imperative mood; try rephrasing", So I'll try rephrasing this.

File: Bio/SeqIO/Interfaces.py [hunk 6]
```diff
@@ -175,19 +175,27 @@ def __init__(self, handle):
         self._footer_written = False
 
     def write_header(self):
-        """Set _header_written to True.
+        """Write the header, method for writing file header.
```
Comment (peterjc) [inline Bio/SeqIO/Interfaces.py]: That's needlessly repetitive, how about: "Write the file header." here?

File: Bio/SeqIO/Interfaces.py [hunk 7]
```diff
@@ -175,19 +175,27 @@ def __init__(self, handle):
...
 
-        To control the header is written only once.
+        If your file format defines a header, you should implement this method
+        in order to write the header before any of the records.
+
+        The default implementation checks the private attribute ._header_written
+        to ensure the footer is only written once.
```
Comment (peterjc) [inline Bio/SeqIO/Interfaces.py]: footer --> header (was probably my typo in the suggested wording)

File: Bio/SeqIO/Interfaces.py [hunk 8]
```diff
@@ -175,19 +175,27 @@ def __init__(self, handle):
...
         """
         assert not self._header_written, "You have aleady called write_header()"
         assert not self._record_written, "You have aleady called write_record() or write_records()"
         assert not self._footer_written, "You have aleady called write_footer()"
         self._header_written = True
 
     def write_footer(self):
-        """Set _footer_written to True.
+        """Write the footer, method for writing file footer.
```
Comment (peterjc) [inline Bio/SeqIO/Interfaces.py]: --> "Write the file footer."?

**Comment by svalqui** [comment]
I think is ready now.

**Comment by peterjc** [comment]
Thank you - merged despite unrelated TravisCI failure, see #1995

**Pull request merged**